component/core/src/main/java/org/exoplatform/social/core/identity/ProfileFilterListAccess.java

```diff
@@ -201,7 +203,11 @@ public int getSize() throws Exception {
       size = identityStorage.getIdentitiesByFirstCharacterOfNameCount(providerId, profileFilter);
     } else if (profileFilter.isEmpty()) {
       if(profileFilter.getViewerIdentity() == null) {
-        size = identityStorage.getIdentitiesByProfileFilterCount(providerId, profileFilter);
+        if (profileFilter.isExcludeExternal()) {
+          size = identityStorage.getInternalIdentitiesCount(providerId);
+        } else {
+          size = identityStorage.getIdentitiesByProfileFilterCount(providerId, profileFilter);
```

Comment:
Update this method getIdentitiesByProfileFilterCount for both case since profileFilter contains the information

Reply:
Done !